modules/security/src/test/java/org/apache/ignite/internal/processor/security/compute/closure/DistributedClosureRemoteSecurityContextCheckTest.java

```diff
@@ -0,0 +1,311 @@
+    private static final String SRV_FEATURE_TRANSITION = "srv_feature_transition";
+
+    /** Name of client feature transit node. */
+    private static final String CLNT_FEATURE_TRANSITION = "clnt_feature_transition";
+
+    /** Name of server endpoint node. */
+    private static final String SRV_ENDPOINT = "srv_endpoint";
+
+    /** Name of client endpoint node. */
+    private static final String CLNT_ENDPOINT = "clnt_endpoint";
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        startGrid(SRV_INITIATOR, allowAllPermissionSet());
+
+        startGrid(CLNT_INITIATOR, allowAllPermissionSet(), true);
+
+        startGrid(SRV_FEATURE_CALL, allowAllPermissionSet());
+
+        startGrid(CLNT_FEATURE_CALL, allowAllPermissionSet(), true);
+
+        startGrid(SRV_FEATURE_TRANSITION, allowAllPermissionSet());
+
+        startGrid(CLNT_FEATURE_TRANSITION, allowAllPermissionSet(), true);
+
+        startGrid(SRV_ENDPOINT, allowAllPermissionSet());
+
+        startGrid(CLNT_ENDPOINT, allowAllPermissionSet(), true);
+
+        G.allGrids().get(0).cluster().active(true);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void setupVerifier(Verifier verifier) {
+        verifier
+            .add(SRV_FEATURE_CALL, 1)
+            .add(CLNT_FEATURE_CALL, 1)
+            .add(SRV_FEATURE_TRANSITION, 2)
+            .add(CLNT_FEATURE_TRANSITION, 2)
+            .add(SRV_ENDPOINT, 4)
+            .add(CLNT_ENDPOINT, 4);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void test() {
+        runAndCheck(grid(SRV_INITIATOR));
+        runAndCheck(grid(CLNT_INITIATOR));
+    }
+
+    /**
+     * @param initiator Initiator node.
+     */
+    private void runAndCheck(IgniteEx initiator) {
+        for (ComputeInvoke ci : ComputeInvoke.values()) {
+            runAndCheck(secSubjectId(initiator),
+                () -> compute(initiator, featureCalls()).broadcast((IgniteRunnable)
+                    new CommonClosure(ci, featureTransitions(), endpoints())
+                )
+            );
+        }
+    }
+
+    /**
+     * @return Collection of feature call nodes ids.
+     */
+    private Collection<UUID> featureCalls() {
+        return Arrays.asList(
+            nodeId(SRV_FEATURE_CALL),
+            nodeId(CLNT_FEATURE_CALL)
+        );
+    }
+
+    /**
+     * @return Collection of feature transit nodes ids.
+     */
+    private Collection<UUID> featureTransitions() {
+        return Arrays.asList(
+            nodeId(SRV_FEATURE_TRANSITION),
+            nodeId(CLNT_FEATURE_TRANSITION)
+        );
+    }
+
+    /**
+     * @return Collection of endpont nodes ids.
+     */
+    private Collection<UUID> endpoints() {
+        return Arrays.asList(
+            nodeId(SRV_ENDPOINT),
+            nodeId(CLNT_ENDPOINT)
+        );
+    }
+
+    /** Enum for ways to invoke compute. */
+    enum ComputeInvoke {
```

Comment:
Can we use `Consumer` instead of enum with a switch?

Reply:
We can use consumers for certain compute methods but it would be bulky because we must use here only classes rather then lambdas.

Reply:
done